python/nav/web/auth.py

```diff
@@ -14,17 +13,87 @@
 # License along with NAV. If not, see <http://www.gnu.org/licenses/>.
 #
 """
-Contains web authentication functionality for NAV.
+Contains web authentication and login functionality for NAV.
+
+The "*Middleware" is Django-specific.
 """
+
+from datetime import datetime
 import logging
+from os.path import join
+import os
 
-from nav.web import ldapauth
+from django.conf import settings
+from django.contrib.sessions.backends.db import SessionStore
+from django.http import HttpResponseRedirect, HttpResponse
+from django.urls import reverse
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.six.moves.urllib import parse
+
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:  # Django <= 1.9
+    MiddlewareMixin = object
+
+try:
+    # Python 3.6+
+    import secrets
+
+    def fake_password(length):
+        return secrets.token_urlsafe(length)
+
+except ImportError:
+    from random import choice
+    import string
+
+    def fake_password(length):
+        symbols = string.ascii_letters + string.punctuation + string.digits
+        return u"".join(choice(symbols) for i in range(length))
+
+
+from nav.auditlog.models import LogEntry
+from nav.config import NAVConfigParser
+from nav.django.utils import is_admin, get_account
 from nav.models.profiles import Account, AccountGroup
+from nav.web import ldapauth
 
 
 _logger = logging.getLogger(__name__)
 
 
+ACCOUNT_ID_VAR = 'account_id'
+SUDOER_ID_VAR = 'sudoer'
+
+# This may seem like redundant information, but it seems django's reverse
+# will hang under some usages of these middleware classes - so until we figure
+# out what's going on, we'll hardcode this here.
+LOGIN_URL = '/index/login/'
+# The local logout url, redirects to '/' after logout
+# If the entire site is protected via remote_user, this link must be outside
+# that protection!
+LOGOUT_URL = '/index/logout/'
+
+
+class RemoteUserConfigParser(NAVConfigParser):
+    DEFAULT_CONFIG_FILES = [join('webfront', 'webfront.conf')]
+    DEFAULT_CONFIG = u"""
+[remote-user]
+enabled=no
+login-url=
+logout-url=
```

Comment:
These two options are neither documented in the reference docs, or in the supplied example `webfront.conf`